crates/core/src/yarn_fn/function_wrapping.rs

```diff
@@ -132,6 +143,13 @@ impl PartialEq for Box<dyn UntypedYarnFn + Send + Sync> {
 
 impl Eq for Box<dyn UntypedYarnFn + Send + Sync> {}
 
+#[macro_export]
+macro_rules! yarn_fn {
+    (($($param:ty),+) -> $ret:ty) => {
+        impl YarnFn<fn($($param),+) -> $ret, Out = $ret>
```

Comment:
Will passing `fn` instead of `FnMut` (which isn't possible due to nested `impl`) limit the usability of this macro?